src/core/public/fatal_errors/fatal_errors_service.test.ts

```diff
@@ -0,0 +1,130 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as Rx from 'rxjs';
+
+expect.addSnapshotSerializer({
+  test: val => val instanceof Rx.Observable,
+  print: () => `Rx.Observable`,
+});
+
+const mockRender = jest.fn();
+jest.mock('react-dom', () => {
+  return {
+    render: mockRender,
+  };
+});
+
+import { FatalErrorsService } from './fatal_errors_service';
+
+function setup() {
+  const rootDomElement = document.createElement('div');
+
+  const injectedMetadata = {
+    getKibanaBuildNumber: jest.fn().mockReturnValue('kibanaBuildNumber'),
+    getKibanaVersion: jest.fn().mockReturnValue('kibanaVersion'),
+  };
+
+  const stopCoreSystem = jest.fn();
+
+  return {
+    rootDomElement,
+    injectedMetadata,
+    stopCoreSystem,
+    fatalErrors: new FatalErrorsService({
+      injectedMetadata,
+      rootDomElement,
+      stopCoreSystem,
+    } as any),
+  };
+}
+
+afterEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('#add()', () => {
+  it('calls stopCoreSystem() param', () => {
+    const { stopCoreSystem, fatalErrors } = setup();
+
+    expect(stopCoreSystem).not.toHaveBeenCalled();
+    expect(() => {
+      fatalErrors.add(new Error('foo'));
+    }).toThrowError();
+    expect(stopCoreSystem).toHaveBeenCalled();
+    expect(stopCoreSystem).toHaveBeenCalledWith();
+  });
+
+  it('deletes all children of rootDomElement and renders <FatalErrorScreen /> into it', () => {
+    const { fatalErrors, rootDomElement } = setup();
+
+    rootDomElement.innerHTML = `
+      <h1>Loading...</h1>
+      <div class="someSpinner"></div>
+    `;
+
+    expect(mockRender).not.toHaveBeenCalled();
+    expect(rootDomElement.children).toHaveLength(2);
+    expect(() => {
+      fatalErrors.add(new Error('foo'));
+    }).toThrowError();
+    expect(mockRender).toHaveBeenCalledTimes(1);
+    expect(rootDomElement.children).toHaveLength(1);
+    expect(rootDomElement.children[0]).toMatchSnapshot('fatal error screen container');
+    expect(mockRender.mock.calls[0][0]).toMatchSnapshot('fatal error screen component');
+  });
+});
+
+describe('start.add()', () => {
+  it('exposes a function that passes its two arguments to fatalErrors.add()', () => {
+    const { fatalErrors } = setup();
+
+    jest.spyOn(fatalErrors, 'add').mockImplementation(() => {
+      /* noop */
+    });
+
+    expect(fatalErrors.add).not.toHaveBeenCalled();
+    const { add } = fatalErrors.start();
+    add('foo', 'bar');
```

Comment:
It's trying to check that the `add()` method on the start contract calls the `add()` method on the service. It would fail if the add method was updated and just logged the errors, didn't send them to the service for some reason. We know that the methods are exactly the same, but the test is designed to adapt to changes to the fatal errors service down the road.